src/neo-vm/Collections/OrderedDictionary.cs

```diff
@@ -0,0 +1,112 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Neo.VM.Collections
+{
+    internal class OrderedDictionary<TKey, TValue> : IDictionary<TKey, TValue>
+    {
+        private class TItem
+        {
+            public TKey Key;
+            public TValue Value;
+        }
+
+        private class InternalCollection : KeyedCollection<TKey, TItem>
```

Comment:
This is the faster one?

Reply:
I just copied the code from https://github.com/neo-project/neo/blob/master/src/neo/IO/Caching/OrderedDictionary.cs

It is deterministic. We need it in VM.

Reply:
I prefer to use SortedDictionary, but we can do benchmark later

Reply:
`SortedDictionary` will sort the keys, which is not needed for us, and there will be additional overhead. We just need to arrange the items in the order they were added.